timesketch/api/v1/resources/settings.py

```diff
@@ -27,46 +15,78 @@ class SystemSettingsResource(Resource):
     @login_required
     def get(self):
         """GET system settings.
-
         Returns:
             JSON object with system settings.
         """
-        # Settings from timesketch.conf to expose to the frontend clients.
-        settings_to_return = ["DFIQ_ENABLED"]
-        result = {}
-
-        for setting in settings_to_return:
-            result[setting] = current_app.config.get(setting)
+        # Settings from timesketch.conf to expose to the frontend clients
+        result = {
+            "DFIQ_ENABLED": current_app.config.get("DFIQ_ENABLED"),
+            "SEARCH_PROCESSING_TIMELINES": current_app.config.get(
+                "SEARCH_PROCESSING_TIMELINES", False
+            ),
+        }
 
-        # Derive the default LLM provider from the new configuration.
-        # Expecting the "default" config to be a dict with exactly one key:
-        # the provider name.
         llm_configs = current_app.config.get("LLM_PROVIDER_CONFIGS", {})
-        default_provider = None
-        default_conf = llm_configs.get("default")
-        if default_conf and isinstance(default_conf, dict) and len(default_conf) == 1:
-            default_provider = next(iter(default_conf))
-        result["LLM_PROVIDER"] = default_provider
+        result["LLM_PROVIDER"] = bool(llm_configs)
 
-        # TODO(mvd): Remove by 2025/06/01 once all users have updated their config.
-        old_llm_provider = current_app.config.get("LLM_PROVIDER")
+        # Check default LLM provider
+        default_provider_working = False
```

Comment:
I wonder if we should move checks for configs working into a separate private functions of the system settings class? This would improve maintainability and readability I think.

Reply:
Yep fantastic suggestion, good example of me just keeping coding and not being patient enough to take a step back.